core.md

```diff
@@ -177,84 +249,105 @@ initialized to 0, fields of type `any` are initialized to `undefined`,
+};
+const RectangleType = new StructType({topLeft: PointType, bottomRight: PointType},
+                                   {defaults: defaults});
+// Instantiate from a source object with one partially and one entirely missing field:
+let rect1 = new RectangleType({topLeft: {x: 10}});
+rect1.topLeft.x === 10;
+rect1.topLeft.y === Number.NEGATIVE_INFINITY;
+rect1.bottomRight.x === Number.POSITIVE_INFINITY;
+rect1.bottomRight.y === Number.POSITIVE_INFINITY;
 ```
 
-The result will be two objects as shown:
+It's an error to provide default values with incorrect types for members with primitive
+type. For members with struct type, only the structure of the default value (and the
+types of any fields that provide default values for members with primitive type contained
+in the struct type) is relevant, not its type.
 
-    line1 ----> +===========+
-                | buffer    | ----> +============+ ArrayBuffer
-                | offset: 0 |       | from: x: 1 |
-                +===========+       |       y: 2 |
-                                    | to:   x: 3 |
-                                    |       y: 4 |
-                                    +============+
+As with builtin default values, overridden default values only apply during
+instantiation, not during assignment. That is, it's still an error to assign an
+incomplete example object to a member that is an embedded struct.
 
-As you can see from the diagram, the typed object `line1` doesn't
-actually store any data itself. Instead, it is simply a pointer into a
-backing store (an array buffer, same as from typed arrays) that
-contains the data itself.
+### Instantiating struct type arrays
 
-*Efficiency note:* The spec has been designed so that, most of the
-time, engines only have to create the backing buffer object and not
-the pointer object `line1`. Instead of creating an object to store the
-buffer and offset, the engine can usually just store the buffer and
-offset directly as synthetic local variables.
+For each type definition, a fixed-sized typed array of instances of that type can be
+created using the type's `.array` constructor.
+
+Just as typed array constructors, typed object array constructors support four
+different overloads:
+
+```js
+const PointType = new StructType({x: float64, y: float64});
+// Creates an instance of length 10, with all entries initialized to default values.
+let points = new PointType.array(10);
+// Creates a copy of `points`.
+let pointsCopy = new PointType.array(points);
+// Creates an instance by iterating over the array-like or iterable source and
+// creating instances of `PointType` for all encountered items.
+let coercedPoints = new PointType.array([new PointType(1, 2), new PointType(10, 20)]);
+// Creates an instance as a view onto the given buffer, starting at the given
+// offset and with the given length, both of which are optional.
+let pointsView = new PointType.array(buffer(points), 3, 3);
```

Comment:
I changed this to introduce canonicalization between struct arrays. However, I'm somewhat on the fence on this. While I agree with your intuition that the previous behavior would be somewhat surprising, I'm not sure how bad that'd be. The same set of equivalences (and lack thereof) holds for `let a = [{}, {}]; let b = a.slice();`

One could argue that it *would* also hold for current typed arrays, except it's not observable because they can only contain primitive values.

I think it wouldn't actually matter too much one way or the other, tbh. Except with array canonicalization, we shut the door on making typed object arrays a generalization of the current typed arrays, including respecifying them in terms of typed object arrays. I'm also not sure how crucial this would be, but it'd at least be very nice.